osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs

```diff
@@ -131,6 +177,18 @@ private void selectedItemChanged()
             UpdateMods();
 
             Ruleset.Value = SelectedItem.Value.Ruleset.Value;
+
+            if (SelectedItem.Value?.AllowedMods.Any() != true)
```

Comment:
`SelectedItem.Value` cannot be null at this stage due to the early null guard a few lines up. Therefore this won't hide the mod overlay if the item changes to `null`.

Reply:
i think the null check may not be required then? i noticed in testing that the room doesn't go null when no selection exists, but is a placeholder empty model instead.

Reply:
The selected item goes null for a short time when room settings change. This is to prevent showing an outdated beatmap while the online lookup happens.